src/extension_commands.ml

```diff
@@ -950,6 +950,220 @@ module Search_by_type = struct
+        current_position
+        (results : Range.t list)
+        text_editor
+        client
+        instance
+    =
+    let open Promise.Syntax in
+    let selected_item = ref None in
+    let text_document = TextEditor.document text_editor in
+    let quickPickItems =
+      List.map results ~f:(fun res ->
+        let line = Position.line @@ Range.end_ res in
+        QuickPickItem.create
+          ~label:(Printf.sprintf "Line %d" line)
+          ~detail:
+            (Printf.sprintf
+               "%s"
+               (TextLine.text @@ TextDocument.lineAt ~line text_document))
+          ~description:(Range.json_of_t res |> Jsonoo.stringify)
+          ())
+    in
+    let module QuickPick = Vscode.QuickPick.Make (QuickPickItem) in
+    let quickPick =
+      QuickPick.set
+        (Window.createQuickPick (module QuickPickItem) ())
+        ~title:"Typed Holes"
+        ~activeItems:[]
+        ~busy:false
+        ~enabled:true
+        ~placeholder:"Use arrow keys to preview / Select to jump to it"
+        ~selectedItems:[]
+        ~ignoreFocusOut:true
+        ~items:quickPickItems
+        ~buttons:[]
+        ()
+    in
+    let _disposable =
+      QuickPick.onDidChangeSelection
+        quickPick
+        ~listener:(fun selection ->
+          match selection with
+          | item :: _ ->
+            let range_string = QuickPickItem.description item in
+            (match range_string with
+             | Some r ->
+               (match Jsonoo.try_parse_opt r with
+                | Some range ->
+                  let range = Range.t_of_json range in
+                  move_to_hole range text_editor
+                | None -> ())
+             | _ -> ())
+          | _ -> ())
+        ()
+    in
+    let _disposable =
+      QuickPick.onDidChangeActive
+        quickPick
+        ~listener:(fun selection ->
+          match selection with
+          | item :: _ ->
+            let range_string = QuickPickItem.description item in
+            (match range_string with
+             | Some r ->
+               (match Jsonoo.try_parse_opt r with
+                | Some range ->
+                  let range = Range.t_of_json range in
+                  move_to_hole range text_editor
+                | None -> ())
+             | _ -> ())
+          | _ -> ())
+        ()
+    in
+    let _disposable =
+      QuickPick.onDidAccept
+        quickPick
+        ~listener:(fun () ->
+          match QuickPick.selectedItems quickPick with
+          | Some (item :: _) ->
+            let range_string = QuickPickItem.description item in
+            ignore
+              (match range_string with
+               | Some r ->
+                 (match Jsonoo.try_parse_opt r with
+                  | Some range ->
+                    let range = Range.t_of_json range in
+                    let* () = jump_to_hole range text_editor in
+                    selected_item := Some range;
+                    QuickPick.hide quickPick;
```

Comment:
I would expect the quickpick to hide without the need for this.

Reply:
for some reason this is not the behavior I observe. It doesn't close if I don't manually hide it.